templates.go

```diff
@@ -394,15 +429,32 @@ write_files:
                 path: /readiness
                 port: 8081
                 scheme: HTTP
-              initialDelaySeconds: 30
+              initialDelaySeconds: 3
               timeoutSeconds: 5
           - name: dnsmasq
-            image: gcr.io/google_containers/kube-dnsmasq-amd64:1.4
+            image: gcr.io/google_containers/k8s-dns-dnsmasq-nanny-amd64:1.14.2
+            livenessProbe:
+              httpGet:
+                path: /healthcheck/dnsmasq
+                port: 10054
+                scheme: HTTP
+              initialDelaySeconds: 60
+              timeoutSeconds: 5
+              successThreshold: 1
+              failureThreshold: 5
             args:
+            - -v=2
+            - -logtostderr
+            - -configDir=/etc/k8s/dns/dnsmasq-nanny
+            - -restartDnsmasq=true
+            - --
```

Comment:
should be right https://github.com/kubernetes/kubernetes/blob/master/cluster/addons/dns/kubedns-controller.yaml.base#L124
and this is also what I tried on cluster zero